### PR TITLE
Booking app

### PR DESCRIPTION
**Pull request by angelikaMierzejewska**

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -1,11 +1,24 @@
...
+import { SearchComponent } from './modules/search/containers/search/search.component';
+import { ProfileComponent } from './modules/user/containers/profile/profile.component';
+import { AuthGuard } from './modules/user/guard/auth.guard';
+import { HotelComponent } from './modules/search/containers/hotel/hotel.component';
+import { BookingComponent } from './modules/search/containers/booking/booking.component';
 
-
-const routes: Routes = [];
+const routes: Routes = [
+  { path: '', component: SearchComponent },
```
Comment (figuratively) [inline src/app/app-routing.module.ts]: this should be at the end of this list

File: src/app/containers/tool-bar/tool-bar.component.html [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        <button mat-button [matMenuTriggerFor]="menu">PL</button>
+        <mat-menu #menu="matMenu">
+          <button mat-menu-item>PL</button>
+          <button mat-menu-item>EN</button>
+        </mat-menu>
+      </div>
+
+      <ng-template #logout>
+        <div fxShow="true" fxHide.lt-md class="navbar navbar--web">
+          <button mat-button routerLink="">Search</button>
```
Comment (figuratively) [inline src/app/containers/tool-bar/tool-bar.component.html]: empty routerLink

File: src/app/containers/tool-bar/tool-bar.component.html [hunk 3]
```diff
@@ -0,0 +1,56 @@
+<div>
+  <mat-toolbar color="primary">
+    <mat-toolbar-row>
+      <span>My app</span>
+
+      <div fxShow="true" fxHide.lt-md class="navbar navbar--web" *ngIf="user$ | async; else logout">
+        <button mat-button routerLink="">Search</button>
```
Comment (figuratively) [inline src/app/containers/tool-bar/tool-bar.component.html]: empty routerLink

File: src/app/modules/search/containers/hotel/hotel.component.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+})
+export class HotelComponent implements OnInit {
+  private hotel: Hotel;
+  constructor(private hotelService: HotelService, private route: ActivatedRoute) {}
+
+  ngOnInit() {
+    this.getHotelData();
+  }
+
+  getHotelData(): void {
```
Comment (figuratively) [inline src/app/modules/search/containers/hotel/hotel.component.ts]: add access modifier

File: src/app/modules/search/containers/rooms/rooms.component.html [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<div *ngFor="let room of rooms$ | async">
+  <p>
+    Room number: {{ room.number }}, capacity {{ room.capacity }}
+    <a mat-raised-button color="primary" (click)="onBooking(room.id)">Book</a>
```
Comment (figuratively) [inline src/app/modules/search/containers/rooms/rooms.component.html]: you should use button instead

File: src/app/modules/search/containers/rooms/rooms.component.ts [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+  ngOnInit() {
+    this.roomService.getAllRooms().subscribe();
+
+    this.rooms$ = this.store
+      .select<Room[]>('rooms')
+      .pipe(map((rooms: Room[]) => rooms.filter((room: Room) => room.hotel.id === this.hotelId)));
+  }
+
+  onBooking(id: number) {
```
Comment (figuratively) [inline src/app/modules/search/containers/rooms/rooms.component.ts]: add an access modifier and the type of the returned value

File: src/app/modules/user/containers/login/login.component.html [hunk 7]
```diff
@@ -0,0 +1,115 @@
+<mat-dialog-content>
+  <mat-card class="user-form__box" [ngClass]="{ 'user-form__box--height': registerForm }">
```
Comment (figuratively) [inline src/app/modules/user/containers/login/login.component.html]: user-form__box--height is not defined in SCSS

File: src/app/modules/user/containers/login/login.component.html [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+      class="user-form__img"
+      src="https://valueadd.pl//themes/contrib/valueadd/assets/img/valueadd-logo.png"
+    />
+    <form [formGroup]="loginForm" (ngSubmit)="submit()">
+      <mat-card-content>
+        <mat-form-field *ngIf="registerForm">
+          <input type="firstName" matInput placeholder="First Name" formControlName="firstName" />
+          <mat-error
+            *ngIf="
+              loginForm.get('firstName').hasError('required') && loginForm.get('firstName').touched
```
Comment (figuratively) [inline src/app/modules/user/containers/login/login.component.html]: functions shouldn't be performed in HTML

File: src/app/modules/user/containers/login/login.component.html [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+            "
+          >
+            This field is required
+          </mat-error>
+        </mat-form-field>
+        <mat-form-field *ngIf="registerForm">
+          <input type="lastName" matInput placeholder="Last Name" formControlName="lastName" />
+          <mat-error
+            *ngIf="
+              loginForm.get('lastName').hasError('required') && loginForm.get('lastName').touched
```
Comment (figuratively) [inline src/app/modules/user/containers/login/login.component.html]: functions shouldn't be performed in HTML

File: src/app/modules/user/containers/login/login.component.html [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+          >
+            This field is required
+          </mat-error>
+        </mat-form-field>
+        <mat-form-field *ngIf="registerForm">
+          <input type="imageUrl" matInput placeholder="Image url" formControlName="imageUrl" />
+        </mat-form-field>
+        <mat-form-field *ngIf="registerForm">
+          <mat-icon
+            *ngIf="loginForm.get('email').valid"
```
Comment (figuratively) [inline src/app/modules/user/containers/login/login.component.html]: functions shouldn't be performed in HTML

File: src/app/modules/user/containers/login/login.component.html [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+            matSuffix
+            class="user-form__icon login__icon--done"
+            >done</mat-icon
+          >
+          <mat-icon
+            *ngIf="
+              (loginForm.get('email').hasError('required') && loginForm.get('email').touched) ||
```
Comment (figuratively) [inline src/app/modules/user/containers/login/login.component.html]: functions shouldn't be performed in HTML

File: src/app/modules/user/containers/login/login.component.html [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+              loginForm.get('email').hasError('email')
+            "
+            matSuffix
+            class="user-form__icon--error"
+          >
+            clear
+          </mat-icon>
+          <input type="email" matInput placeholder="Email" formControlName="email" />
+          <mat-error
+            *ngIf="loginForm.get('email').hasError('required') && loginForm.get('email').touched"
```
Comment (figuratively) [inline src/app/modules/user/containers/login/login.component.html]: functions shouldn't be performed in HTML

File: src/app/modules/user/containers/login/login.component.html [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+          >
+            This field is required
+          </mat-error>
+          <mat-error
+            *ngIf="loginForm.get('email').hasError('email') && loginForm.get('email').touched"
```
Comment (figuratively) [inline src/app/modules/user/containers/login/login.component.html]: functions shouldn't be performed in HTML

File: src/app/modules/user/containers/login/login.component.html [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+          >
+            Invalid email format.
+          </mat-error>
+        </mat-form-field>
+
+        <mat-form-field>
+          <input type="login" matInput placeholder="User name" formControlName="login" />
+          <mat-error
+            *ngIf="loginForm.get('login').hasError('required') && loginForm.get('login').touched"
```
Comment (figuratively) [inline src/app/modules/user/containers/login/login.component.html]: functions shouldn't be performed in HTML

File: src/app/modules/user/containers/login/login.component.html [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+          >
+            This field is required
+          </mat-error>
+        </mat-form-field>
+
+        <mat-form-field>
+          <mat-icon
+            *ngIf="loginForm.get('password').valid"
```
Comment (figuratively) [inline src/app/modules/user/containers/login/login.component.html]: functions shouldn't be performed in HTML

File: src/app/modules/user/containers/login/login.component.html [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+            matSuffix
+            class="user-form__icon login__icon--done"
+            >done</mat-icon
+          >
+          <mat-icon
+            *ngIf="
+              loginForm.get('password').hasError('required') && loginForm.get('password').touched
```
Comment (figuratively) [inline src/app/modules/user/containers/login/login.component.html]: functions shouldn't be performed in HTML

File: src/app/modules/user/containers/login/login.component.html [hunk 17]
```diff
@@ -0,0 +1,115 @@
...
+            "
+            matSuffix
+            class="user-form__icon--error"
+          >
+            clear
+          </mat-icon>
+          <input type="password" matInput placeholder="Password" formControlName="password" />
+          <mat-error
+            *ngIf="
+              loginForm.get('password').hasError('required') && loginForm.get('password').touched
```
Comment (figuratively) [inline src/app/modules/user/containers/login/login.component.html]: functions shouldn't be performed in HTML

File: src/app/modules/user/containers/login/login.component.ts [hunk 18]
```diff
@@ -0,0 +1,75 @@
...
+    firstName: [''],
+    lastName: [''],
+    email: ['', [Validators.email]],
+    login: ['', [Validators.required, Validators.maxLength(16)]],
+    password: [
+      '',
+      [
+        Validators.required,
+        Validators.minLength(8)
+        // Validators.pattern('(.*[A-Z].*)')
```
Comment (figuratively) [inline src/app/modules/user/containers/login/login.component.ts]: remove this comment

File: src/app/modules/user/containers/login/login.component.html [hunk 19]
```diff
@@ -0,0 +1,115 @@
...
+          <mat-icon
+            *ngIf="loginForm.get('email').valid"
+            matSuffix
+            class="user-form__icon login__icon--done"
+            >done</mat-icon
+          >
+          <mat-icon
+            *ngIf="
+              (loginForm.get('email').hasError('required') && loginForm.get('email').touched) ||
+              loginForm.get('email').hasError('email')
```
Comment (figuratively) [inline src/app/modules/user/containers/login/login.component.html]: functions shouldn't be performed in HTML

File: src/app/modules/search/services/facalities.services.ts [hunk 20]
```diff
@@ -0,0 +1,20 @@
...
+
+@Injectable({
+  providedIn: 'root'
+})
+export class FacalitiesServices {
+  private urlBase = 'http://185.157.80.88:8080';
+
+  constructor(private httpClient: HttpClient, private store: Store) {}
+
+  getFacalities(): Observable<any> {
```
[the remark this answers is not in the document]
Comment (adrianzaorski) [reply src/app/modules/search/services/facalities.services.ts]: Type that Observable.

**Pull request merged**